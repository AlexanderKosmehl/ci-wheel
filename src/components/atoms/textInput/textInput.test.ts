import {
  describe, expect, it, vi,
} from 'vitest';
import generateTextInput from './textInput';

describe('generateButton', () => {
  const onKeyPress = vi.fn((_event: KeyboardEvent) => {});

  const inputBar = generateTextInput({
    classes: ['class', 'class2'],
    placeholder: 'Placeholder',
    onKeyPress,
  });

  it('generates component correctly', () => {
    expect(inputBar).toMatchSnapshot();
  });

  it('adds working callback', () => {
    inputBar.value = 'Test';
    inputBar.dispatchEvent(new Event('keyup'));

    expect(onKeyPress).toBeCalled();
  });
});
