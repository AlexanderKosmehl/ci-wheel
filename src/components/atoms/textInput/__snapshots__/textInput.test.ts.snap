// Vitest Snapshot v1

exports[`generateButton > generates component correctly 1`] = `
<input
  class="class class2 _input_6ed02a"
  placeholder="Placeholder"
  type="text"
/>
`;
